GitHub Workflow: support uploading build report from Windows

https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793

```
Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    RELEASE_VERSION: 
    OSSRH_USERNAME: 
    OSSRH_PASSWORD: 
    ORG_GRADLE_PROJECT_signingKey: 
    ORG_GRADLE_PROJECT_signingPassword: 
    GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx1[2](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:2)g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
    GRADLE_ACTION_ID: gradle/actions/setup-gradle
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    JAVA_HOME_17_X64: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
    GRADLE_BUILD_ACTION_CACHE_RESTORED: true
    DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
    DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
    GITHUB_DEPENDENCY_GRAPH_ENABLED: false
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.Run find . -type d -name 'reports' | zip -@ -r build-reports.zip
  find . -type d -name 'reports' | zip -@ -r build-reports.zip
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    RELEASE_VERSION: 
    OSSRH_USERNAME: 
    OSSRH_PASSWORD: 
    ORG_GRADLE_PROJECT_signingKey: 
    ORG_GRADLE_PROJECT_signingPassword: 
    GRADLE_OPTS: -Dorg.gradle.jvmargs="-Xmx1[2](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:2)g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8"
    GRADLE_ACTION_ID: gradle/actions/setup-gradle
    JAVA_HOME: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    JAVA_HOME_17_X64: C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\17.0.11-9\x64
    GRADLE_BUILD_ACTION_SETUP_COMPLETED: true
    GRADLE_BUILD_ACTION_CACHE_RESTORED: true
    DEVELOCITY_INJECTION_INIT_SCRIPT_NAME: gradle-actions.inject-develocity.init.gradle
    DEVELOCITY_AUTO_INJECTION_CUSTOM_VALUE: gradle-actions
    GITHUB_DEPENDENCY_GRAPH_ENABLED: false
zip: D:\a\_temp\71c08419-1c19-4c[3](https://github.com/kotest/kotest/actions/runs/9482603753/job/26128051793#step:7:3)e-b0c6-fdbb3508ae53.ps1:2
Line |
   2 |  find . -type d -name 'reports' | zip -@ -r build-reports.zip
     |                                   ~~~
     | The term 'zip' is not recognized as a name of a cmdlet, function, script file, or executable program. Check the
     | spelling of the name, or if a path was included, verify that the path is correct and try again.
Error: Process completed with exit code 1.
```

Maybe set the shell to bash explicitly?

--- a/.github/workflows/run-gradle.yml
+++ b/.github/workflows/run-gradle.yml
@@ -44,17 +44,16 @@ jobs:
             run: ./gradlew ${{ inputs.task }} --scan
             shell: bash
 
-         -  name: Bundle the build report
-            if: failure()
-            run: find . -type d -name 'reports' | zip -@ -r build-reports.zip
-            shell: bash
-
-         -  name: Upload the build report
-            if: failure()
-            uses: actions/upload-artifact@v4
-            with:
-               name: error-report
-               path: build-reports.zip
+         - name: Upload build reports
+           if: failure()
+           uses: actions/upload-artifact@v4
+           with:
+              name: build-reports-${{ runner.os }}-${{ github.action }}-${{ github.run_id }}
+              path: |
+                 **/build/reports/
+                 **/*.hprof
+                 **/*.log
+              if-no-files-found: ignore
 
 env:
    RELEASE_VERSION: ${{ github.event.inputs.version }}

